src/main/java/co/cask/PluginJSON.java

```diff
@@ -321,31 +331,34 @@ private String getDataURISchemaAsString(File file, String scheme, String mediaTy
   /**
    * Initializes this Mojo, extracts all necessary paths and configurations.
    */
-  private void initialize() {
+  private void initialize() throws MojoExecutionException {
     groupId = project.getGroupId();
     artifactId = project.getArtifactId();
     version = project.getVersion();
     baseDirectory = project.getBasedir();
     buildDirectory = new File(project.getBuild().getDirectory());
-    if (!widgetsDirectory.contains("/")) {
-      widgetDirectory = new File(project.getBasedir() + "/" + widgetsDirectory);
-    } else {
-      widgetDirectory = new File(widgetsDirectory);
-    }
+    widgetDirectory = createDirectory(baseDirectory, widgetsDirectory);
+    iconDirectory = createDirectory(baseDirectory, iconsDirectory);
+    docDirectory = createDirectory(baseDirectory, docsDirectory);
+  }
 
-    if (iconsDirectory != null) {
-      if (!iconsDirectory.contains("/")) {
-        iconDirectory = new File(project.getBasedir() + "/" + iconsDirectory);
-      } else {
-        iconDirectory = new File(iconsDirectory);
-      }
+  /**
+   * Create directories based on given project
+   */
+  private File createDirectory(File baseDirectory, String newDirectoryName) throws MojoExecutionException {
+    File createdDirectory;
+    if (newDirectoryName.startsWith("/")) {
+      createdDirectory = new File(newDirectoryName);
+    } else {
+      createdDirectory = new File(baseDirectory, newDirectoryName);
     }
 
-    if (!docsDirectory.contains("/")) {
-      docDirectory = new File(project.getBasedir() + "/" + docsDirectory);
-    } else {
-      docDirectory = new File(docsDirectory);
+    if (!(createdDirectory.exists() && createdDirectory.isDirectory())) {
+      throw new MojoExecutionException(
+        String.format("Unable to create directory '%s'.", newDirectoryName)
```

Comment:
this error message is misleading, you're not creating the directory. If it doesn't exist, throw an exception about it not existing. If it's not a directory, throw an exception about it not being a directory.